File: lovenotes-frontend/src/app/services/love-note.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { LoveNote } from '../models/love-note.model';

@Injectable({
  providedIn: 'root'
})
export class LoveNoteService {
  private apiUrl = 'http://localhost:8080/api/lovenotes';

  constructor(private http: HttpClient) { }

  getAllLoveNotes(): Observable<LoveNote[]> {
    return this.http.get<LoveNote[]>(this.apiUrl);
  }

  getLoveNoteById(id: number): Observable<LoveNote> {
    return this.http.get<LoveNote>(`${this.apiUrl}/${id}`);
  }

  createLoveNote(loveNote: LoveNote): Observable<LoveNote> {
    return this.http.post<LoveNote>(this.apiUrl, loveNote);
  }

  updateLoveNote(id: number, loveNote: LoveNote): Observable<LoveNote> {
    return this.http.put<LoveNote>(`${this.apiUrl}/${id}`, loveNote);
  }

  deleteLoveNote(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: lovenotes/src/main/java/com/valentine/lovenotes/service/LoveNoteService.java
package com.valentine.lovenotes.service;

import com.valentine.lovenotes.model.LoveNote;
import com.valentine.lovenotes.repository.LoveNoteRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class LoveNoteService {
    
    @Autowired
    private LoveNoteRepository loveNoteRepository;
    
    // Create a new love note
    public LoveNote createLoveNote(LoveNote loveNote) {
        return loveNoteRepository.save(loveNote);
    }
    
    // Get all love notes
    public List<LoveNote> getAllLoveNotes() {
        return loveNoteRepository.findAll();
    }
    
    // Get love note by ID
    public Optional<LoveNote> getLoveNoteById(Long id) {
        return loveNoteRepository.findById(id);
    }
    
    // Update love note
    public LoveNote updateLoveNote(Long id, LoveNote updatedNote) {
        return loveNoteRepository.findById(id)
            .map(note -> {
                note.setMessage(updatedNote.getMessage());
                note.setAuthorName(updatedNote.getAuthorName());
                note.setIsAnonymous(updatedNote.getIsAnonymous());
                return loveNoteRepository.save(note);
            })
            .orElseThrow(() -> new RuntimeException("Love note not found with id: " + id));
    }
    
    // Delete love note
    public void deleteLoveNote(Long id) {
        loveNoteRepository.deleteById(id);
    }
}
